src/base/searchengine.cpp

```diff
@@ -240,6 +240,19 @@ bool SearchEngine::uninstallPlugin(const QString &name)
     return true;
 }
 
+void SearchEngine::updateIconPath(PluginInfo * const plugin)
```

Comment:
1. If this method should modify any plugin info (even for not installed plugin) turn it into `static` since it doesn't access any non-static members. Also you should use Q_ASSERT(plugin).
2. If this method should modify only installed plugins you should change its signature to accept plugin name and get plugin info itself. Since this method is public we can't trust the passed plugin info is valid.